### DISCUSSION TITLE
How to use variables/ locals in dependencies

### DISCUSSION THREAD

**Discussion by gitPlatformTests** [gruntwork-io/knowledge-base#793] [How do I]
Hey,
I have declared a root terragrunt.hcl file with:
```
locals {
  modules_path = "/path/to/modules"
```

in my chile terragrunt.hcl file i did:
```
dependency "blah" {
  config_path = "${local.modules_path}//path/to/dependency"
}
```

is it possible to set in the parent configuration a path and use it in dependencies in child modules?
thanks

---

<ins datetime="2023-12-14T10:27:04Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110643">Tracked in ticket #110643</a></p>
</ins>

**Comment by gitsstewart** [comment]
@gitPlatformTests have you already looked through the docs to see if the examples and guidance there answers your question? https://terragrunt.gruntwork.io/docs/features/keep-your-terragrunt-architecture-dry/#using-read_terragrunt_config-to-dry-parent-configurations

**Reply by gitPlatformTests** [reply]
yeah
also found this
https://terragrunt.gruntwork.io/docs/rfc/imports/

not sure if this is up to date or not

**Comment by denis256** [comment]
Hello,
can be accessed include locals in childs through `include.root.locals.<name>`

```
# parent terragrunt.hcl
locals {
  config_path = "${get_repo_root()}/discussion-793/module"
}

# child terragrunt.hcl
include "root" {
  path   = find_in_parent_folders()
  expose = true
}

dependency "blah" {
  config_path = "${include.root.locals.config_path}"
  skip_outputs = true
}
```
https://terragrunt.gruntwork.io/docs/reference/config-blocks-and-attributes/#include
https://github.com/denis256/terragrunt-tests/tree/master/discussion-793
